consensus/abci.go

```diff
@@ -45,11 +49,14 @@ type TendermintABCI struct {
 	db           *database.Database
 	checkTxState *state.StateDB
 	ethRPCClient *rpc.Client
-	logger       tmtLog.Logger
-	metrics      metrics.Metrics
 
-	getCurrentDBState func() (*state.StateDB, error)
-	getCurrentBlock   func() *ethTypes.Block
+	logger         tmtLog.Logger
+	metrics        metrics.Metrics
+	curBlockHeader *tmtAbciTypes.Header
```

Comment:
I am not modifying this object, but it might be modified internally by Tendermint since it is taken till the value `curBlockHeader.Proposer` is used. Therefore IMO we need the pointer in this case.

Reply:
I see hmmhm in this case though, the header shouldn't really change right? If it does, that's no bueno as we already executed some business logic relying on that information.

I investigated this a bit more and actually, Tendermint won't modify it because this is how Tendermint sends the object.

```
func (abci *TendermintABCI) BeginBlock(req tmtAbciTypes.RequestBeginBlock)
```

... without a pointer. You created the pointer manually yourself hehe:

```
abci.curBlockHeader = &req.Header
```